config-controller/internal/controller/policy_controller.go

```diff
@@ -0,0 +1,134 @@
+	"github.com/stackrox/rox/roxctl/common/printer"
+
+	configstackroxiov1alpha1 "github.com/stackrox/rox/config-controller/api/v1alpha1"
+)
+
+// PolicyReconciler reconciles a Policy object
+type PolicyReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	conn   *grpc.ClientConn
+}
+
+//+kubebuilder:rbac:groups=config.stackrox.io,resources=policies,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=config.stackrox.io,resources=policies/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=config.stackrox.io,resources=policies/finalizers,verbs=update
+
+func (r *PolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	rlog := log.FromContext(ctx)
+	rlog.Info("Reconciling", "namespace", req.Namespace, "name", req.Name)
+
+	// Get the policy CR
+	policyCR := &configstackroxiov1alpha1.Policy{}
+	if err := r.Client.Get(ctx, req.NamespacedName, policyCR); err != nil {
+		if k8serr.IsNotFound(err) {
+			// Must have been deleted
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("Failed to get policy: namespace=%s, name=%s", req.Namespace, req.Name)
+	}
+
+	desiredState := &storage.Policy{}
+	if err := protojson.Unmarshal([]byte(policyCR.Spec.Policy), protoadapt.MessageV2Of(desiredState)); err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "Failed to parse policy JSON")
+	}
+
+	// GET policy from Central
+	svc := v1.NewPolicyServiceClient(r.conn)
+	allPolicies, err := svc.ListPolicies(ctx, &v1.RawQuery{})
+
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "Failed to list all policies")
+	}
+
+	for _, policy := range allPolicies.Policies {
+		if policy.Name == req.Name {
+			desiredState.Id = policy.Id
+			if _, err = svc.PutPolicy(ctx, desiredState); err != nil {
+				return ctrl.Result{}, errors.Wrap(err, fmt.Sprintf("Failed to PUT policy %s", req.Name))
+			}
+			policyCR.Status.Accepted = true
+			policyCR.Status.Message = "Successfully updated policy"
+			if err = r.Client.Status().Update(ctx, policyCR); err != nil {
+				return ctrl.Result{}, errors.Wrap(err, fmt.Sprintf("Failed to set status on policy %s", req.Name))
+			}
+			return ctrl.Result{}, nil
+		}
+	}
+
+	if _, err = svc.PostPolicy(ctx, &v1.PostPolicyRequest{Policy: desiredState}); err != nil {
+		wrappedErr := errors.Wrap(err, fmt.Sprintf("Failed to create policy %s", req.Name))
+		policyCR.Status.Accepted = false
+		policyCR.Status.Message = wrappedErr.Error()
+		if err = r.Client.Status().Update(ctx, policyCR); err != nil {
+			return ctrl.Result{}, errors.Wrap(err, fmt.Sprintf("Failed to set status on policy %s", req.Name))
+		}
+		return ctrl.Result{}, wrappedErr
+	} else {
+		policyCR.Status.Accepted = true
+		policyCR.Status.Message = "Successfully created policy"
+		if err = r.Client.Status().Update(ctx, policyCR); err != nil {
+			return ctrl.Result{}, errors.Wrap(err, fmt.Sprintf("Failed to set status on policy %s", req.Name))
+		}
+	}
+
+	// Report status
```

Comment:
??

Reply:
Another leftover comment from when I stubbed out the function.